server/document.js

```diff
@@ -97,11 +97,14 @@ export class NextScript extends Component {
 
   render () {
```

Comment:
Not critical or a blocker for this approach to work, but I would love to see the addition of `<link rel="preload">` to see if helps mitigate the added latency of downloading the components as script tags.